lmp/bb-config.sh

```diff
@@ -236,11 +240,6 @@ ARCHIVER_MODE[src] = "original"
 ARCHIVER_MODE[diff] = "1"
 EOFEOF
 
-if [ $(ls ../sstate-cache | wc -l) -ne 0 ] ; then
-	status "Found existing sstate cache, using local copy"
-	echo 'SSTATE_MIRRORS = ""' >> conf/auto.conf
-fi
```

Comment:
Can you elaborate what this change is for in the commit message, I don't quite understand this. 

I'd think this specific line of code would be needed or our customer factory builds would break?

Reply:
I am assuming this is the local cache of the factory. That just doesn't exist in the first build, being populated after this so this `../sstate-cache` folder always have some content. Then the sstate cache will always be off after the first build.

Reply:
I need to look on the job server code to see what is mounted on this `../sstate-cache` folder

Reply:
I have checked this snip `Found existing sstate cache, using local copy` and it is used always after the first build.
So it is what I am think about, our factories only use the mirror for the first build and after `SSTATE_MIRRORS = ""`.
I will update to reflect that.

Reply:
Actually I think this is not correct and we should always use SSTATE_MIRRORS so the factories can take advantage of the public cache when updating from v(N-1) to v(N).

Bitbake will check for sstate cache in the following order:

1. SSTATE_DIR
2. SSTATE_MIRRORS

So it will only search the mirror if the file not found on the local cache `../sstate-cache`, this is an insignificant overhead searching the `SSTATE_MIRRORS` because most of the files are on the local `SSTATE_DIR`

Reply:
So should we change to always use the http mirror by default? Currently an elif.

Reply:
that's about right. When using NFS (aka "persistent jobserv volumes"), we don't want to use HTTP because it causes duplicate copies of all the base LmP objects. Makes each factory about 7G more multiplied by `num_customers`.

for our "nocache" builds (like the LmP) they don't use persistent volumes, so HTTP works pretty well. *and* our public LmP builds are done on CI workers that copy the sstate cache generated locally back to the storage bucket (where HTTP is served from).

Reply:
Right, so with this PR we add the options to change the default mirror used in the factory.

By default the mirror used is the local one that is the shared NFS mounted on the container:
https://github.com/foundriesio/ci-scripts/blob/f56cc8b494a5882a30471100d44b95bcd605f038/lmp/bb-config.sh#L20

The http mirror is disabled in costumer factories and is enabled only on public lmp with:
https://github.com/foundriesio/ci-scripts/blob/f56cc8b494a5882a30471100d44b95bcd605f038/lmp/jobserv.yml#L626

With the above same change in any costumer factory the user can change the mirror

Reply:
the only missing part is the increment of the `LMP_VERSION_CACHE` for the development release. Maybe we need the same `LMP_VERSION_CACHE_DEV` on ci too.
https://github.com/foundriesio/ci-scripts/blob/f56cc8b494a5882a30471100d44b95bcd605f038/helpers.sh#L97-L100

Reply:
Also add the `LMP_VERSION_CACHE_DEV` so everything is in place to use the HTTP mirror on the factories